navigation/libandroid-navigation/src/main/java/com/mapbox/services/android/navigation/v5/MapboxNavigation.java

```diff
@@ -438,7 +455,7 @@ public void onServiceConnected(ComponentName componentName, IBinder service) {
       if (alertLevelChangeListeners != null) {
         navigationService.setAlertLevelChangeListeners(alertLevelChangeListeners);
       }
-
+      progressChangeListeners.add(MapboxNavigation.this);
```

Comment:
This doesn't seem right, the object is implementing a listener that is keeping track of. Also, the next line is always true because there always is at least one ProgressChangeListener. Could you elaborate why is this needed?